plugins/module_utils/proxmox.py

```diff
@@ -170,6 +189,13 @@ def get_pool(self, poolid):
         except Exception as e:
             self.module.fail_json(msg="Unable to retrieve pool %s information: %s" % (poolid, e))
 
+    def get_storage(self, storage):
+        try:
+            storage = self.proxmox_api.storage.get(storage)
+        except Exception:
+            self.module.fail_json(msg="Storage '%s' does not exist" % storage)
```

Comment:
I guess the exception could be caused by other reasons as well. Not reporting it to the user (so it's possible to understand when something unexpected triggered "Storage xxx does not exist") seems wrong to me.